ksql-functional-tests/src/test/resources/query-validation-tests/aslist.json

```diff
@@ -0,0 +1,20 @@
+{
+  "comments": [
+    "Tests covering AS_LIST functionality with generics"
+  ],
+  "tests": [
+    {
+      "name": "construct a list from two elements",
+      "statements": [
+        "CREATE STREAM TEST (a INT, b INT) WITH (kafka_topic='test_topic', value_format='JSON');",
```

Comment:
how about when `(a INT, b BIGINT)`  - what happens then? Should add a test to cover that.

Reply:
as it stands, this fails since they're considered different types. implementing this functionality should be possible (I actually have an idea on how to do [this](https://en.wikipedia.org/wiki/Fermat%27s_Last_Theorem) pretty easily but I'll leave that for a future PR)